src/RewardsStreamerMP.sol

```diff
@@ -161,6 +155,11 @@ contract RewardsStreamerMP is IStakeManager, TrustedCodehashAccess, ReentrancyGu
         // Adjust rewardIndex before updating totalMP
         uint256 previousTotalWeight = totalStaked + totalMP;
         totalMP += accruedMP;
+
+        // FIXME: If newTotalWeight > previousTotalWeight, then (previousTotalWeight) / (newTotalWeight) is less than 1.
+        // Multiplying rewardIndex by a fraction less than 1 reduces its value.
+        // Users who have not updated their accountRewardIndex may see a reduction in their pending rewards.
+        // Possible fix: adjust only when newTotalWeight < previousTotalWeight
```

Comment:
@gravityblast I've been pondering on this comment for quite a while now and I can't quite understand what the issue here is. Can you give a concrete example that demonstrates the issue you're describing here?